internal/cloud/realm/user.go

```diff
@@ -92,11 +93,139 @@ type User struct {
 // UserIdentity is a Realm application user identity
 type UserIdentity struct {
 	UID          string                 `json:"id"`
-	ProviderType string                 `json:"provider_type"`
+	ProviderType ProviderType           `json:"provider_type"`
 	ProviderID   primitive.ObjectID     `json:"provider_id"`
 	ProviderData map[string]interface{} `json:"provider_data,omitempty"`
 }
 
+// ProviderType is a Realm application provider type
+type ProviderType string
+
+// set of supported provider type values
+const (
+	ProviderTypeEmpty          ProviderType = ""
+	ProviderTypeUserPassord    ProviderType = "local-userpass"
+	ProviderTypeAPIKey         ProviderType = "api-key"
+	ProviderTypeFacebook       ProviderType = "oauth2-facebook"
+	ProviderTypeGoogle         ProviderType = "oauth2-google"
+	ProviderTypeAnonymous      ProviderType = "anon-user"
+	ProviderTypeCustomToken    ProviderType = "custom-token"
+	ProviderTypeApple          ProviderType = "oauth2-apple"
+	ProviderTypeCustomFunction ProviderType = "custom-function"
+)
+
+// slice of supported provider type values as string slice
+var (
+	ValidProviderTypes = []string{
+		ProviderTypeUserPassord.String(),
+		ProviderTypeAPIKey.String(),
+		ProviderTypeFacebook.String(),
+		ProviderTypeGoogle.String(),
+		ProviderTypeAnonymous.String(),
+		ProviderTypeCustomToken.String(),
+		ProviderTypeApple.String(),
+		ProviderTypeCustomFunction.String(),
+	}
+)
+
+// String returns the provider type string
+func (pt ProviderType) String() string { return string(pt) }
+
+// Display returns the provider type display string
+func (pt ProviderType) Display() string {
+	switch pt {
+	case ProviderTypeAnonymous:
+		return "Anonymous"
+	case ProviderTypeUserPassord:
+		return "User/Password"
+	case ProviderTypeAPIKey:
+		return "ApiKey"
+	case ProviderTypeApple:
+		return "Apple"
+	case ProviderTypeGoogle:
+		return "Google"
+	case ProviderTypeFacebook:
+		return "Facebook"
+	case ProviderTypeCustomToken:
+		return "Custom JWT"
+	case ProviderTypeCustomFunction:
+		return "Custom Function"
+	default:
+		return "Unknown"
+	}
+}
+
+// IsValid validates the provided provider type
+func (pt ProviderType) IsValid() error {
+	switch pt {
+	case ProviderTypeAnonymous,
+		ProviderTypeUserPassord,
+		ProviderTypeAPIKey,
+		ProviderTypeApple,
+		ProviderTypeGoogle,
+		ProviderTypeFacebook,
+		ProviderTypeCustomToken,
+		ProviderTypeCustomFunction:
+		return nil
+	default:
+		return errors.New("Invalid ProviderType")
+	}
+}
+
+// DisplayUser returns the display string for a user and provider type
+func (pt ProviderType) DisplayUser(user User) string {
```

Comment:
any chance these "user display" functions (here and below) could live in the `commands/user` package as just unexported helper functions (assuming that's where they're used)?